### DISCUSSION TITLE
Undefiend Property in One to Many Relationship

### DISCUSSION THREAD

**Discussion by JeremieConstant** [Laravel-Backpack/community-forum#639] [Q&A (Help)] [answered]
Hello dear community,

im building a course management application based on Laravel Backback **Pro**
I had an already Working Event Model in Default Laravel.

Many Events can have one course. One Course can have many events.

```php
<?php

namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use Backpack\CRUD\app\Models\Traits\HasIdentifiableAttribute;
use Carbon\Carbon;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use App\Models\Course;

class Event extends Model
{
    use HasFactory, CrudTrait, HasIdentifiableAttribute;

    protected $connection = 'mysql_typo3';
    protected $table = 'tx_cbwcourse_domain_model_events';
    protected $primaryKey = 'uid';
    protected $with = ['course', 'coursetype', 'place'];

    public function course(): BelongsTo
    {
        return $this->belongsTo(Course::class, 'course', 'uid');
    }

    public function coursetype(): BelongsTo
    {
        return $this->belongsTo(Coursetype::class, 'coursetype', 'uid');
    }

    public function place(): BelongsTo
    {
        return $this->belongsTo(Place::class, 'location', 'uid');
    }
}
```
In Default Laravel i can access the courses information from my event model like this:
`{{ $event->getRelationValue('course')->title }}`

When using Backpack i get this error:
`Undefined property: App\Models\Event::$coursetype`
![grafik](https://github.com/Laravel-Backpack/community-forum/assets/12912800/ad24608c-c1f2-4633-9b49-b07836b108d9)

I used `php artisan backpack:crud event` to create everything and this error came up.
I tried to change the type from "text" to "relationship" and all kinds of scenarios, but i can't seem to fix that error.

Anyone have a clue on how to solve this?

**Comment by tabacitu** [comment]
@JeremieConstant if the Eloquent relationships are defined, Backpack should pick them up just fine. So I suspect there's some misconfiguration here, either in the relationships in the Model, or in the CrudController regarding the fields/columns. What does your `setupListOperation()` look like?

**Reply by JeremieConstant** [reply, marked as answer]
@tabacitu thanks for following up on my topic. I could find a solution to the problem, which i wanted to share, since it seems to be a backpack related issue.

In my "Event" model the field "course" holds the uid of the related Course model.
I noticed that when i rename the "course" function which resolves the relationship to something else, backpack no longer throws and error and correctly resolved the relation.

Before:
`public function course(): BelongsTo
    {
        return $this->belongsTo(Course::class, 'course', 'uid');
    }
`

After:
`public function coursefoo(): BelongsTo
    {
        return $this->belongsTo(Course::class, 'course', 'uid');
    }
`

I assume the function to resolve the relation to a model must not have the same name as the model itself.
Is this a new discovery or covered by some paradigms which i've overlooked?

**Reply by tabacitu** [reply]
Hmmm interesting. This might happen if Backpack is confused about what `course` is. Do you happen to also have a `course` attribute on the model? Or something like that? If so, being specific about the field and column _type_ would fix it. But you already mentioned making the type `relationship`... that would have been my first solution too 🤷‍♂️
